go/private/test.bzl

```diff
@@ -59,45 +59,32 @@ def _go_test_impl(ctx):
       env = dict(go_toolchain.env, RUNDIR=ctx.label.package)
   )
 
-  if "race" not in ctx.features:
-    emit_go_compile_action(
-      ctx,
-      sources=depset([main_go]),
-      libs=[golib.library],
-      lib_paths=[golib.searchpath],
-      direct_paths=[golib.importpath],
-      out_object=main_object,
-      gc_goopts=get_gc_goopts(ctx),
-    )
-    emit_go_pack_action(ctx, main_lib, [main_object])
-    emit_go_link_action(
-      ctx,
-      transitive_go_library_paths=golib.transitive_go_library_paths,
-      transitive_go_libraries=golib.transitive_go_libraries,
-      cgo_deps=golib.transitive_cgo_deps,
-      libs=[main_lib],
-      executable=ctx.outputs.executable,
-      gc_linkopts=gc_linkopts(ctx),
-      x_defs=ctx.attr.x_defs)
-  else:
-    emit_go_compile_action(
-      ctx,
-      sources=depset([main_go]),
-      libs=[golib.race],
-      lib_paths=[golib.searchpath_race],
-      direct_paths=[golib.importpath],
-      out_object=main_object,
-      gc_goopts=get_gc_goopts(ctx) + ["-race"],
-    )
-    emit_go_pack_action(ctx, main_lib, [main_object])
-    emit_go_link_action(
+  main_lib, _ = emit_library_actions(ctx,
+      srcs = [main_go],
+      deps = [],
+      cgo_object = None,
+      library = None,
+      want_coverage = False,
+      importpath = ctx.label.name + "~testmain~",
+      golibs = [golib],
+  )
+
+  go_library_paths=golib.transitive_go_library_paths
```

Comment:
Nit: space around = in this block

Also, consider making this an if-else, rather than just an if. I'm a little uncomfortable with `go_libraries` and `go_library_paths` being assigned more than once along the same path.